pytorch_lightning/trainer/properties.py

```diff
@@ -223,8 +224,13 @@ def model(self, model: torch.nn.Module) -> None:
     def log_dir(self) -> Optional[str]:
         if self.logger is None:
             dirpath = self.default_root_dir
+        elif isinstance(self.logger, TensorBoardLogger):
+            dirpath = self.logger.log_dir
         else:
-            dirpath = getattr(self.logger, 'log_dir' if isinstance(self.logger, TensorBoardLogger) else 'save_dir')
+            dirpath = self.logger.save_dir
+
+        if isinstance(self.logger, LoggerCollection) and dirpath is None:
+            dirpath = self.default_root_dir
```

Comment:
Why not move this if to the `if-elif` chain above?

Reply:
Its condition was a little more specific than the others, so I thought it would make more sense for it to be separate. On second thought, though, letting users override the default behavior by manually setting the `save_dir` of their LoggerCollection probably isn't very clean. I've removed the second condition and merge the blocks.